lib/broadway/processor.ex

```diff
@@ -69,9 +69,11 @@ defmodule Broadway.Processor do
       kind, error ->
         Logger.error(Exception.format(kind, error, System.stacktrace()))
         message = Message.failed(message, "due to an unhandled #{kind}")
+        message = maybe_handle_failed_message(message, module, context)
```

Comment:
Instead of calling it per message, I would call it right before line 48 above. Then I would also recommend moving `maybe_handle_failed_message` to a `@doc false` function in `Broadway.Acknowledger`.

Reply:
Done.